docs/design/build-system.md

```diff
@@ -0,0 +1,154 @@
+# Build System
+
+Angular CLI includes a first-party build system for Angular applications distributed as `@angular-devkit/build-angular`.
+This build system is responsible for creating a standalone application from user source files and third-party dependencies.
+
+`@angular-devkit/build-angular` itself integrates with the rest of Angular CLI by being an [Architect builder](https://angular.io/guide/cli-builder).
+This document describes a top level view of the functionality in `@angular-devkit/build-angular`, referred as just the "build system".
+Deprecated or soon to be removed features are not described here.
+
+In broad strokes the main areas are:
+- loading and processing sources
+- code splitting
+- production optimizations
+- post-processing steps
+
+Many tools are used in this process, and most of these steps happen within a [Webpack](https://webpack.js.org/) build.
+We maintain a number of Webpack-centric plugins in this repository, some of these are public but most are private since they are very specific to our setup.
+
+
+## Loading and processing sources
+
+Sources for Angular CLI browser apps are comprised of TypeScript files, style sheets, assets, scripts, and third party dependencies.
+A given build will load these sources from disk, process them, and bundle them together.
+
+
+### TypeScript
+
+Angular builds rely heavily on TypeScript-specific functionality for Ahead-of-Time template compilation (AOT).
+Outside Angular CLI, this is performed by the Angular Compiler (`ngc`), provided by `@angular/compiler-cli`.
+To avail of Ahead-of-Time template compilation within a Webpack compilation we use and distribute the `@ngtools/webpack` Webpack plugin.
+
+Typescript sources are loaded from disk and compiled in-memory into Javascript files that are stored in a virtual file system and made available to Webpack.
+During compilation we also perform a number of code transformations using TypeScript transformers that enable automatic usage of AOT, internationalization features, and server-side rendering.
+
+AOT compilation requires loading HTML and CSS resources referenced on Angular Components. These are processed using a Webpack child compilation.
```

Comment:
Is there any definition or explanation of what a "child compilation" is and why it is used for HTML and CSS assets? I encountered this earlier, but I couldn't find any real documentation and didn't fully grasp the purpose behind it.

Reply:
Does this change help? I couldn't find webpack docs for them.

> AOT compilation requires loading HTML and CSS resources, referenced on Angular Components, as standalone strings with no external dependencies.
However, Webpack compilations operate on the basis of modules and references between them.
To obtain the standalone string we compile resources using a separate Webpack child compilation then extract the results.
These child compilations inherit configuration and access to the same files as the parent compilation, but have their own compilation life cycle and complete independently.

Reply:
What exactly is the purpose of a child compilation here? What benefit to we get from using these child compilations as opposed to just doing the same thing in the parent compilation?